addons/stock_dropshipping/stock_dropshipping.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, api, _
+from openerp.exceptions import Warning
+
+class stock_warehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    @api.model
+    def _get_drop_shipping_route(self):
+        try:
+            ds_route_id = self.env.ref("stock_dropshipping.route_drop_shipping")
+        except:
+            ds_route_id = self.env['stock.location.route'].search([('name', 'like', _('Drop Shipping'))])
+            ds_route_id = ds_route_id and ds_route_id[0] or False
+        if not ds_route_id:
+            raise Warning(_('Can\'t find any generic Drop Shipping route.'))
+        return ds_route_id.id
+
+class sale_order_line(models.Model):
+    _inherit = 'sale.order.line'
+
```

Comment:
The inheritance in stock_dropshipping might be overkill.  Maybe I suggested this, but it is better to just check if the route_id on the sale order line has a rule with a location_id of usage customer and this procurement rule is a 'buy' rule.